drivers/wifi/siwx917/Kconfig.siwx917

```diff
@@ -32,6 +32,37 @@ config WIFI_SIWX917_NET_STACK_OFFLOAD
 
 endchoice
 
+config WIFI_SIWX917_AP_MODE
+	bool "Enable AP Mode"
+	help
+	 Enable this option to allow the device to operate in
+	 Access Point (AP) mode instead of the default Station (STA) mode.
+
+config WIFI_SIWX917_MAX_STATIONS
+	int "Maximum number of stations supported by AP"
+	range 1 8
+	default 4
+	depends on WIFI_SIWX917_AP_MODE
+	help
+	  Sets the maximum number of stations that the
+	  Access Point (AP) can support.
+
+config WIFI_SIWX917_BEACON_INTERVAL
+	int "Beacon interval (in milliseconds)"
+	range 100 1000
+	default 100
+	depends on WIFI_SIWX917_AP_MODE
+	help
+	  Defines the interval between beacon transmissions in AP mode.
+
+config WIFI_SIWX917_DTIM_BEACON_COUNT
+	int "DTIM Beacon Count"
+	default 3
+	depends on WIFI_SIWX917_AP_MODE
+	help
+	 Specifies the number of beacon intervals that must elapse
+	 before the AP sends buffered multicast/broadcast traffic.
```

Comment:
An implementation of AP may provides tens of tuning options. Do we want to expose all of them in Kconfig? In addition, I feel it make sense (at least it does not consume more memory) to set these value during runtime.

So, maybe it is better to hardcode sane values and leave the opportunity to the user to patch the code if he really want to do something else?

I think it is not a blocker for now, because the number of options is limited. However, I would take care of the inflation of the number of options.